packages/squiggle-lang/src/rescript/Distributions/GenericDist/GenericDist.res

```diff
@@ -207,15 +207,17 @@ module AlgebraicCombination = {
     | Some(Ok(symbolicDist)) => Ok(Symbolic(symbolicDist))
     | Some(Error(e)) => Error(Other(e))
     | None =>
-      switch chooseConvolutionOrMonteCarlo(t1, t2) {
-      | #CalculateWithMonteCarlo => runMonteCarlo(toSampleSetFn, arithmeticOperation, t1, t2)
-      | #CalculateWithConvolution =>
-        runConvolution(
-          toPointSetFn,
-          arithmeticOperation,
-          t1,
-          t2,
-        )->E.R2.fmap(r => DistributionTypes.PointSet(r))
+      switch arithmeticOperation {
+      | #Divide
+      | #Power
+      | #Logarithm =>
+        runMonteCarlo(toSampleSetFn, arithmeticOperation, t1, t2)
+      | (#Add | #Subtract | #Multiply) as op =>
+        switch chooseConvolutionOrMonteCarlo(t1, t2) {
```

Comment:
Why not just merge the logic above into the ``chooseConvolutionOrMonteCarlo`` method?

Reply:
The design I chose here was because I wanted to rely on the type system. This design means that passing invalid operations to use convolution cannot happen as the type system checks it rather than a check at runtime.

I think this is safer (bug / extension wise)? I'm just following my FP intuition that I should never need to write a case for something that never occurs (relying on just chooseConvolutionOrMonteCarlo would either require handling the case of #Logarithm in the convolution methods, or returning a `convolutionOperation` from `chooseConvolutionOrMonteCarlo`).